qa/tasks/cephfs/test_cephfs_shell.py

```diff
@@ -393,6 +393,21 @@ def test_get_without_target_name(self):
             # if the int() fails then that means there's a valid stat output
             pass
 
+    def test_get_subdir_creation(self):
+        s = 'somedata'
+        self.get_cephfs_shell_cmd_output("mkdir tmp")
+        self.get_cephfs_shell_cmd_output("put - /tmp/dump7", stdin=s)
+        self.get_cephfs_shell_cmd_output("get /tmp/dump7 ./dump7")
+        # if get cmd is creating subdirs on its own then dump7 will be
+        # stored as ./dump7/tmp/dump7 and not ./dump7, therefore
+        # if doing `cat ./dump7` returns non-zero exit code(i.e. 1) then
+        # it implies that no such file exists at that location
+        r = self.mount_a.run_shell("cat ./dump7", cwd=None,
+                                   check_status=False)
+        log.info("cat ./dump7 output:\n{}".format(r.stdout.getvalue()))
+        self.assertEqual(r.returncode, 0,
+                         "file not found at expected location.")
```

Comment:
Since you are making that change doing this will cleaner and shorter -

```suggestion
        with self.assertRaises(CommandFailedError) as e:
            self.mount_a.run_shell("cat ./dump7", cwd=None)
        self.assertEqual(e.exitstatus, 0)
```

Reply:
If the `CommandFailedError` didn't raise then it will cause the test to quit with msg `AssertionError: CommandFailedError not raised`

Reply:
Okay, I see the confusion now. Why set `check_status` to `False` and then assert? Just this will do the job -

```suggestion
        self.mount_a.run_shell("cat ./dump7", cwd=None)
```

If this line fails, the test will fail which is what you want with this test IIUC.

Reply:
right but it would be a lot easy to understand test with assert.

Reply:
pushed